it/src/test/java/org/corfudb/universe/scenario/AddNodeDuringInvalidateLayoutIT.java

```diff
@@ -0,0 +1,157 @@
+package org.corfudb.universe.scenario;
+
+import org.corfudb.runtime.clients.BaseClient;
+import org.corfudb.runtime.clients.BaseHandler;
+import org.corfudb.runtime.clients.LayoutHandler;
+import org.corfudb.runtime.clients.ManagementHandler;
+import org.corfudb.runtime.clients.NettyClientRouter;
+import org.corfudb.runtime.exceptions.NetworkException;
+import org.corfudb.runtime.view.Layout;
+import org.corfudb.universe.GenericIntegrationTest;
+import org.corfudb.universe.UniverseManager.UniverseWorkflow;
+import org.corfudb.universe.group.cluster.docker.DockerCorfuCluster;
+import org.corfudb.universe.node.client.LocalCorfuClient;
+import org.corfudb.universe.node.server.CorfuServer;
+import org.corfudb.universe.scenario.fixture.Fixture;
+import org.corfudb.universe.universe.UniverseParams;
+import org.corfudb.util.CFUtils;
+import org.corfudb.util.NodeLocator;
+import org.corfudb.util.Sleep;
+import org.junit.Test;
+import org.junit.jupiter.api.Assertions;
+import org.mockito.Mockito;
+
+import java.time.Duration;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class AddNodeDuringInvalidateLayoutIT extends GenericIntegrationTest {
+
+    /**
+     * Test add node behavior while the layout is being invalidated concurrently.
+     * <p>
+     * 1) Deploy and bootstrap a three nodes cluster.
+     * 2) Remove a node.
+     * 2) Try resetting a node and invalidating layout at the same time. This should throw an exception.
+     * 3) Create a separate router for resets. Try resetting a node and invalidating layout at the same.
+     *    The reset should go through.
+     * 4) Create a separate router for adding a node, and invoke the add node procedure.
+     *    Keep invalidating a layout every time RPC is called. The add node should succeed.
+     * 5) Verify that the node is present in the new layout.
+     */
+    @Test(timeout = 300000)
+    public void AddNodeDuringInvalidateLayoutTest() {
+        workflow(wf -> {
+
+                    wf.setupDocker(fixture -> {
+                        fixture.getCluster().numNodes(3);
```

Comment:
it is a 3 node cluster by default, don't need to specify it

Reply:
Fixed